tests/zfs-tests/tests/functional/fault/auto_spare_001_pos.ksh

```diff
@@ -0,0 +1,111 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+#
+# Copyright (c) 2017 by Intel Corporation. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/fault/fault.cfg
+
+#
+# DESCRIPTION:
+# Testing Fault Management Agent ZED Logic - Automated Auto-Spare Test when
+# drive is faulted due to io errors.
+#
+# STRATEGY:
+# 1. Create a pool with hot spares
+# 2. Write multiple files to the pool
+# 3. Clear the cache by exporting/importing pool
+# 4. Add io error handler with zinject
+# 5. Read data from pool to generate io errors
+# 6. ZED using the retire agent swaps in a spare for the faulted drive
+# 7. Wait for faulted drive to be replaced with a spare
+# 8. Clear events
+# 9. Clear the status of the faulted drive
+# 10. ZED recognizes the resilver and when it is complete removes the spare
+# 11. Wait for pool health to return to normal.
+#
+
+verify_runnable "both"
+
+function setup
+{
+	truncate -s $SPA_MINDEVSIZE $VDEV_FILES
+}
+
+function cleanup
+{
+	log_must zinject -c all
+	poolexists $TESTPOOL && destroy_pool $TESTPOOL
+	rm -f $VDEV_FILES
+}
+
+log_assert "Testing automated auto-spare FMA test"
+
+log_onexit cleanup
+
+for type in "mirror" "raidz" "raidz2"
+do
+	setup
+
+	log_must zpool create -f $TESTPOOL $type ${TOP_FILES} spare ${SPARE}
+
+	# Add some data to the pool
+	fill_fs "/$TESTPOOL" 20 20 1048576 2
+
+	retval=$?
+	if (( retval == 28 )); then
+		log_note "No space left on device."
+	elif (( retval != 0 )); then
+		log_fail "Unexpected exit: $retval"
+	fi
+	
+
+	guid=$(get_config $TESTPOOL pool_guid)
+	# Clear the caches by exporting and importing
+	log_must zpool export $TESTPOOL
+	log_must zpool import -s $guid -d /var/tmp
+
+	# Clear zpool events
+	log_must zpool events -c $TESTPOOL
```

Comment:
Why clear the zpool events here and below?  This shouldn't have any effect on the test case.

Reply:
It was an artifact from when I was polling for the events to determine when resilver and replace were finished.